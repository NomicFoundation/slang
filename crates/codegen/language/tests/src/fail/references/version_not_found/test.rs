#![allow(unused_crate_dependencies)]

codegen_language_macros::compile!(Language(
    name = Foo,
    root_item = One,
    leading_trivia = Sequence([]),
    trailing_trivia = Sequence([]),
    versions = ["1.0.0"],
    sections = [Section(
        title = "Section One",
        topics = [Topic(
            title = "Topic One",
            items = [
                Struct(
                    name = One,
                    fields = (
                        field_1 = Optional(kind = Terminal([Two]), enabled = From("3.0.0")),
                        field_2 = Optional(kind = Terminal([Two]))
                    )
                ),
                Token(
                    name = Two,
                    definitions = [TokenDefinition(scanner = Atom("two"))]
                )
            ]
        )]
    )]
));

fn main() {}
