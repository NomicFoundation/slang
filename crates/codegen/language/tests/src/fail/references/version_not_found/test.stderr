error: Version '3.0.0' does not exist in the language definition.
  --> src/fail/references/version_not_found/test.rs:17:76
   |
17 |                         field_1 = Optional(reference = Two, enabled = From("3.0.0")),
   |                                                                            ^^^^^^^
